verify/verify.go

```diff
@@ -1158,6 +1160,69 @@ func RawTdxQuote(raw []byte, options *Options) error {
 	return TdxQuote(quote, options)
 }
 
+func getCertFromBytes(data []byte, phrase string) (*x509.Certificate, error) {
+	cert, rem := pem.Decode(data)
+	if cert == nil {
+		return nil, fmt.Errorf("could not parse PEM formatted %q certificate", phrase)
+	}
+	if len(rem) != 0 {
+		return nil, fmt.Errorf("unexpected trailing bytes: %d bytes", len(rem))
+	}
+	if cert.Type != certificateType {
+		return nil, fmt.Errorf(`the %q PEM block type is %q. Expect %q`, phrase, cert.Type, certificateType)
+	}
+	trustedCert, err := x509.ParseCertificate(cert.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("could not interpret DER bytes of %q certificate : %v", phrase, err)
+	}
+	return trustedCert, nil
+}
+
+func getCertFromPath(path string, phrase string) (*x509.Certificate, error) {
+	certBytes, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return getCertFromBytes(certBytes, phrase)
+}
+
+func getTrustedRoots(rot *cpb.RootOfTrust) (*x509.CertPool, error) {
+	if len(rot.CabundlePaths) == 0 && len(rot.Cabundles) == 0 {
+		return nil, nil
+	}
+	result := x509.NewCertPool()
+	for _, path := range rot.CabundlePaths {
+		cert, err := getCertFromPath(path, "trusted")
```

Comment:
See below about AppendCertsFromPEM to avoid the above helper functions.

Reply:
Done